libs/sdk/eslint-config/src/schematics/shared/rules/modify-eslint-config.ts

```diff
@@ -0,0 +1,38 @@
+import { Rule } from '@angular-devkit/schematics';
+
+import { EsLintConfig } from '../types/eslint-config';
+import { readJsonFile } from '../utility/tree';
+import { writeJsonFile } from '../utility/tree';
+
+const ESLINT_CONFIG_NAME = '@skyux-sdk/eslint-config/recommended';
+const ESLINT_CONFIG_PATH = '.eslintrc.json';
+
+export function modifyEsLintConfig(): Rule {
+  return async (tree) => {
+    const esLintConfig = readJsonFile<EsLintConfig>(tree, ESLINT_CONFIG_PATH);
+
+    if (Array.isArray(esLintConfig.overrides)) {
+      // Setup type-checking for ESLint.
+      esLintConfig.parser = '@typescript-eslint/parser';
+      esLintConfig.parserOptions = {
+        project: ['tsconfig.json'],
+        tsconfigRootDir: '.',
+      };
+
+      // Overwrite the 'extends' array with our configuration.
+      for (const override of esLintConfig.overrides) {
```

Comment:
We are we overwriting things they are extending instead of just supplementing it? I could see a consumer still wanting some custom plugin along with our rules. It seems like we could remove ones that we know that our plugin adds - but removing everything seems like it might be a bit of overreach.

Reply:
We talked about this during Flex and decided on doing what Mary's team does, which is overwriting all the other configs with our own.